tests/test_sphinx_autodoc_typehints.py

```diff
@@ -296,4 +296,28 @@ class InnerClass
 
            Parameters:
               **x** (*a.b.c*) – foo
+
+        dummy_module.function_with_typehint_comment(x)
+
+           Function docstring.
+
+           Parameters:
+              **x** (*int*) – foo
+
+           Return type:
+              None
+
+        class dummy_module.ClassWithTypehints(x)
+
+           Class docstring.
+
+           Parameters:
+              **x** -- foo
```

Comment:
I know this looks like any other unresolvable type. I think we could fix that in the least lines of code by *setting* `__annotations__` on the obj, and calling `get_type_hints` again to resolve the names we extracted from the type hint comments... lmk what you think of this approach